standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/StandardizationInterpreter.scala

```diff
@@ -79,35 +76,24 @@ object StandardizationInterpreter {
     }
 
     // TODO: remove when spark-xml handles empty arrays
-    logger.info(s"Step 1.5: Spark-xml hack (handle empty arrays)")
     val df1 = if (inputType.toLowerCase() == "xml") {
-      expSchema.fields.foldLeft(df)({
-        case (accDf: Dataset[Row], field: StructField) => {
-          val newType = SparkXMLHack.hack(field, "", accDf)
-          accDf.withColumn(field.name, newType)
-        }
-
-      })
+      df.select(expSchema.fields.map{
+        field => SparkXMLHack.hack(field, "", df).as(field.name)
+      }: _*)
     } else df
 
     logger.info(s"Step 2: Standardization")
     // step 2 - standardize
-    val std = expSchema.fields.foldLeft(df1)({
-      case (accDf: Dataset[Row], field: StructField) => {
-
-        logger.info(s"Standardizing field: ${field.name}")
-
-        // Do not standardize existing error column
+    val std = df1.select(expSchema.fields.flatMap{
+      field => logger.info(s"Standardizing field: ${field.name}")
         if (field.name == ErrorMessage.errorColumnName) {
-          accDf
+          Some(df1.col(field.name))
```

Comment:
Why use `Some` here? In the `else` you return `Seq`.

Reply:
Indeed, changed to Seq, thnx